src/normalizeOptions.js

```diff
@@ -51,29 +51,42 @@ function normalizeRoot(opts) {
   }
 }
 
-function normalizeAlias(opts) {
-  opts.regExps = [];
-
-  if (opts.alias) {
-    Object.keys(opts.alias)
-      .filter(isRegExp)
-      .forEach((key) => {
-        const parts = opts.alias[key].split('\\\\');
-
-        function substitute(execResult) {
-          return parts
-            .map(part =>
-              part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
-            )
-            .join('\\');
-        }
+function getAliasPair(key, value) {
+  const parts = value.split('\\\\');
+
+  function substitute(execResult) {
+    return parts
+      .map(part =>
+        part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
+      )
+      .join('\\');
+  }
 
-        opts.regExps.push([new RegExp(key), substitute]);
+  return [new RegExp(key), substitute];
+}
 
-        delete opts.alias[key];
-      });
+function normalizeAlias(opts) {
+  if (opts.alias) {
+    const { alias } = opts;
+    const aliasKeys = Object.keys(alias);
+    const nonRegExpAliases = [];
+    const regExpAliases = [];
+
+    aliasKeys.forEach((key) => {
+      if (isRegExp(key)) {
+        regExpAliases.push(
+          getAliasPair(key, alias[key]),
+        );
+      } else {
+        nonRegExpAliases.push(
+          getAliasPair(`^${key}((?:/|).*)`, `${alias[key]}\\1`),
+        );
+      }
+    });
+
+    opts.alias = [...nonRegExpAliases, ...regExpAliases];
```

Comment:
Indeed, one or the other, we'll have an issue if a user expects the order to be kept. With an object, it's not guaranteed as you say. We'll figure out a way later if really needed. For now, it doesn't seem like users need it.